api/apps/v1alpha1/nemo_customizer_types.go

```diff
@@ -0,0 +1,925 @@
+	// NemoCustomizerStatusNotReady indicates that NEMO CustomizerService is not ready
+	NemoCustomizerStatusNotReady = "NotReady"
+	// NemoCustomizerStatusReady indicates that NEMO CustomizerService is ready
+	NemoCustomizerStatusReady = "Ready"
+	// NemoCustomizerStatusFailed indicates that NEMO CustomizerService has failed
+	NemoCustomizerStatusFailed = "Failed"
+
+	// SchedulerTypeVolcano indicates if the scheduler is volcano
+	SchedulerTypeVolcano = "volcano"
+	// SchedulerTypeRunAI indicates if the scheduler is run.ai
+	SchedulerTypeRunAI = "runai"
+)
+
+// NemoCustomizerSpec defines the desired state of NemoCustomizer
+type NemoCustomizerSpec struct {
+	Image          Image                        `json:"image,omitempty"`
+	Command        []string                     `json:"command,omitempty"`
+	Args           []string                     `json:"args,omitempty"`
+	Env            []corev1.EnvVar              `json:"env,omitempty"`
+	Labels         map[string]string            `json:"labels,omitempty"`
+	Annotations    map[string]string            `json:"annotations,omitempty"`
+	NodeSelector   map[string]string            `json:"nodeSelector,omitempty"`
+	Tolerations    []corev1.Toleration          `json:"tolerations,omitempty"`
+	PodAffinity    *corev1.PodAffinity          `json:"podAffinity,omitempty"`
+	Resources      *corev1.ResourceRequirements `json:"resources,omitempty"`
+	Expose         Expose                       `json:"expose,omitempty"`
+	LivenessProbe  Probe                        `json:"livenessProbe,omitempty"`
+	ReadinessProbe Probe                        `json:"readinessProbe,omitempty"`
+	StartupProbe   Probe                        `json:"startupProbe,omitempty"`
+	Scale          Autoscaling                  `json:"scale,omitempty"`
+	Metrics        Metrics                      `json:"metrics,omitempty"`
+
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default:=1
+	Replicas     int    `json:"replicas,omitempty"`
+	UserID       *int64 `json:"userID,omitempty"`
+	GroupID      *int64 `json:"groupID,omitempty"`
+	RuntimeClass string `json:"runtimeClass,omitempty"`
+
+	// CustomizerConfig stores the customizer configuration for training and models
+	// +kubebuilder:validation:MinLength=1
+	CustomizerConfig string `json:"customizerConfig"`
+
+	// Scheduler Configuration
+	Scheduler Scheduler `json:"scheduler,omitempty"`
+
+	// OpenTelemetry Settings
+	OpenTelemetry OTelSpec `json:"otel"`
+
+	// DatabaseConfig stores the database configuration
+	DatabaseConfig DatabaseConfig `json:"databaseConfig"`
+
+	// WandBSecret stores the secret and encryption key for the Weights and Biases service.
+	WandBSecret WandBSecret `json:"wandbSecret"`
+}
+
+// Scheduler defines the configuration for the scheduler
+type Scheduler struct {
+	// Type is the scheduler type (volcano, runai)
+	// +kubebuilder:validation:Enum=volcano;runai
+	// +kubebuilder:default:=volcano
+	Type string `json:"type,omitempty"`
```

Comment:
Nit -- are there additional scheduler configs to be defined later? Otherwise we can just use a string instead of introducing a struct

Reply:
yes, there can be additional ones, this is a Enum currently but we can expand to other schedulers later on.